common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -0,0 +1,188 @@
+    """
+    return datetime.now(UTC()) > end_date if end_date is not None else False
+
+
+def _add_timezone_string(date_time):
+    """
+    Adds 'UTC' string to the end of start/end date and time texts.
+    """
+    return date_time + u" UTC"
+
+
+def course_start_date_is_default(start, advertised_start):
+    """
+    Returns whether a course's start date hasn't yet been set.
+
+    Arguments:
+        start (datetime): The start datetime of the course in question.
+        advertised_start (str): The advertised start date of the course in question.
+    """
+    return advertised_start is None and start == DEFAULT_START_DATE
+
+
+def course_start_datetime_text(start, advertised_start, format_string, ugettext, strftime):
+    """
+    Calculates text to be shown to user regarding a course's start datetime in UTC..
+    Prefers .advertised_start, then falls back to .start.
+
+    Arguments:
+        start (datetime): the start date of a course
+        advertised_start (str): the advertised start date of a course, as a string
+        format_string (str): the date format type, as passed to strftime
+        ugettext (str -> str): a text localization function
+        strftime (datetime, str -> str): a localized string formatting function
+    """
+
+    def datetime_to_string(date_time):
+        """
+        Returns a formatting datetime, appending a timezone string if specified.
+        """
+        result = strftime(date_time, format_string)
+        return (
+            _add_timezone_string(result) if format_string == "DATE_TIME"
+            else result
+        )
+
+    if advertised_start is not None:
+        try:
+            # from_json either returns a Date, returns None, or raises a ValueError
+            parsed_advertised_start = Date().from_json(advertised_start)
+        except ValueError:
+            parsed_advertised_start = None
+        return (
+            datetime_to_string(parsed_advertised_start) if parsed_advertised_start
+            else advertised_start.title()
+        )
+    elif start != DEFAULT_START_DATE:
+        return datetime_to_string(start)
+    else:
+        _ = ugettext
+        # Translators: TBD stands for 'To Be Determined' and is used when a course
+        # does not yet have an announced start date.
+        return _('TBD')
+
+
+def course_end_datetime_text(end, format_string, strftime_localized):
+    """
+    Returns a formatted string for a course's end date or datetime.
+    If end is none, an empty string will be returned.
+
+    Arguments:
+        end (datetime): the end date of a course
+        format_string (str): the date format type, as passed to strftime
+        strftime_localized (datetime, str -> str): a localized string formatting function
+    """
+    if end is None:
+        return ''
+    else:
+        formatted_date = strftime_localized(end, format_string)
+        return formatted_date if format_string == "SHORT_DATE" else _add_timezone_string(formatted_date)
```

Comment:
I like this idea, but it would involve changing the fields of CourseDescriptor, which I was trying to avoid in this PR. If you think it's worth it, though, I'll look into it.